sdk/inc/azure/iot/az_iot_adu_ota.h

```diff
@@ -0,0 +1,284 @@
+/* ADU Agent States */
+#define AZ_IOT_ADU_OTA_AGENT_STATE_IDLE 0
+#define AZ_IOT_ADU_OTA_AGENT_STATE_DEPLOYMENT_IN_PROGRESS 6
+#define AZ_IOT_ADU_OTA_AGENT_STATE_FAILED 255
+
+/* Maximum Number of Files Handled by this ADU Agent */
+#define AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT 10
+#define MAX_INSTRUCTIONS_STEPS 10
+#define MAX_FILE_HASH_COUNT 2
+
+typedef struct
+{
+  az_span provider;
+  az_span name;
+  az_span version;
+} az_iot_adu_ota_update_id;
+
+/*
+ * For reference:
+ * https://docs.microsoft.com/en-us/azure/iot-hub-device-update/device-update-plug-and-play
+ */
+typedef struct
+{
+  az_span manufacturer;
+  az_span model;
+  az_span adu_version;
+  az_span do_version;
+  az_iot_adu_ota_update_id update_id;
+} az_iot_adu_ota_device_information;
+
+typedef struct
+{
+  az_span step_id;
+  int32_t result_code;
+  int32_t extended_result_code;
+  az_span result_details;
+} az_iot_adu_ota_step_result;
+
+typedef struct
+{
+  int32_t result_code;
+  int32_t extended_result_code;
+  az_span result_details;
+  int32_t step_results_count;
+  az_iot_adu_ota_step_result step_results[MAX_INSTRUCTIONS_STEPS];
+} az_iot_adu_ota_install_result;
+
+typedef struct
+{
+  int32_t action;
+  az_span id;
+  az_span retry_timestamp;
+} az_iot_adu_ota_workflow;
+
+typedef struct
+{
+  az_span id;
+  az_span url;
+} az_iot_adu_ota_file_url;
+
+/**
+ * @brief Structure that holds the parsed contents of the OTA
+ *        request in the Plug and Play writable properties sent
+ *        by the ADU service.
+ */
+typedef struct
+{
+  az_iot_adu_ota_workflow workflow;
+  az_span update_manifest;
+  az_span update_manifest_signature;
+  az_iot_adu_ota_file_url file_urls[AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT];
+  uint32_t file_urls_count;
+} az_iot_adu_ota_update_request;
+
+typedef struct
+{
+  az_span device_manufacturer;
+  az_span device_model;
+} az_iot_adu_ota_update_manifest_compatibility;
+
+typedef struct
+{
+  az_span installed_criteria;
+} az_iot_adu_ota_update_manifest_instructions_step_handler_properties;
+
+typedef struct
+{
+  az_span handler;
+  az_span files[AZ_IOT_ADU_OTA_FILE_URL_MAX_COUNT];
+  uint32_t files_count;
+  az_iot_adu_ota_update_manifest_instructions_step_handler_properties handler_properties;
+} az_iot_adu_ota_update_manifest_instructions_step;
+
+typedef struct
+{
+  az_iot_adu_ota_update_manifest_instructions_step steps[MAX_INSTRUCTIONS_STEPS];
+  uint32_t steps_count;
+} az_iot_adu_ota_update_manifest_instructions;
+
+typedef struct
+{
+  az_span id;
+  az_span hash;
+} az_iot_adu_ota_update_manifest_file_hash;
+
+typedef struct
+{
+  az_span id;
+  az_span file_name;
+  uint32_t size_in_bytes;
+  az_iot_adu_ota_update_manifest_file_hash hashes[MAX_FILE_HASH_COUNT];
+  uint32_t hashes_count;
+} az_iot_adu_ota_update_manifest_file;
+
+/**
+ * @brief Structure that holds the parsed contents of the update manifest
+ *        sent by the ADU service.
+ */
+typedef struct
+{
+  az_span manifest_version;
+  az_iot_adu_ota_update_id update_id;
```

Comment:
Nit  - there's a tab on this line.  It looks aligned on GitHub but on editors (like Source Insight 4, who else still uses it in 2022 and loving it?) it shows misaligned.